src/internal/fileutil/file.go

```diff
@@ -0,0 +1,27 @@
+package fileutil
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+func WithTmpFile(prefix string, cb func(*os.File) error) (retErr error) {
+	if err := os.MkdirAll(os.TempDir(), 0700); err != nil {
```

Comment:
I don't think we should be initializing the operating systems tmp directory in a library.  That check should happen during setup.